harness/determined/common/storage/gcs.py

```diff
@@ -60,6 +60,7 @@ def restore_path(self, storage_id: str) -> Iterator[str]:
         finally:
             self._remove_checkpoint_directory(storage_id)
 
+    @no_type_check
```

Comment:
can you explain what this is for?  the [typing docs](https://docs.python.org/3/library/typing.html#typing.no_type_check) say it indicates that 'type annotations are not type hints' but that doesn't make sense to me.